test/boost/sstable_compaction_test.cc

```diff
@@ -121,7 +121,7 @@ class strategy_control_for_test : public strategy_control {
     }
 
     std::vector<sstables::shared_sstable> candidates(table_state& t) const override {
-        return _candidates_opt.value_or(boost::copy_range<std::vector<sstables::shared_sstable>>(*t.main_sstable_set().all()));
+        return _candidates_opt.value_or(*t.main_sstable_set().all() | std::ranges::to<std::vector<sstables::shared_sstable>>());
```

Comment:
std::vector's type explicitly written. In other places it is deduced.

Reply:
will drop the element type. as the return type is right in the function's signature.